thrimshim/thrimshim/main.py

```diff
@@ -219,15 +220,25 @@ def reset_row(ident):
 def main(connection_string, host='0.0.0.0', port=8004, backdoor_port=0):
 	"""Thrimshim service."""
 	server = WSGIServer((host, port), cors(app))
-	app.db_manager = database.DBManager(dsn=connection_string)
 
+	app.no_authentication = no_authentication
+	app.db_manager = None
+	stopping = gevent.event.Event()
```

Comment:
`stopping` isn't being set anywhere, so it does nothing.
You should move the `def stop()` bit to before connecting to the DB, and add `stopping.set()` to the `stop()` function.